### ISSUE TITLE
Odd behaviour when USERNAME_FIELD is email

### ISSUE THREAD

**Issue by filipeximenes** [omab/python-social-auth#360]
I'm setting `email` as the `USERNAME_FIELD` in my custom auth user.  
This is causing psa to add provider username in the django user email.  
If I keep `'social.pipeline.user.user_details'``in the pipeline, this problem is not perceptible, since it will set the proper email before saving the user. I do need this in my pipeline and I also want to use   
``SOCIAL_AUTH_PROTECTED_USER_FIELDS = ['email',]```  
 setting, so the email is only saved on the first time the user is created.   

To sum up, if I use `SOCIAL_AUTH_PROTECTED_USER_FIELDS = ['email',]`, my user email field is going to be setted as a common username (not a email) by the end of the pipeline.
